### DISCUSSION TITLE
FDS Compilation

### DISCUSSION THREAD

**Discussion by lihy213** [firemodels/fds#12649] [Q&A]
Dear contributors,

I encountered some problems when I compiled FDS source codes. I installed the Intel oneAPI Base and HPC Toolkits, and the Fortran code ran well. In addition, the GNU make and git were also installed. But when I typed `make_fds.bat`, it came to the following errors:
![image](https://github.com/firemodels/fds/assets/96673442/ec64f675-711d-42d3-8192-032bb71f4840)

My OS is *Windows 11*.
What can I do to solve this problem?

Kind regards,

LI

**Comment by mcgratta** [comment]
We are currently transitioning from the old Fortran compiler (ifort) to the new one (ifx). We have not thoroughly tested the new compiler, but try the following. In the `makefile`, change the word `ifort` to `ifx`
```
# set compiler type

I_IFORT=ifort
ifdef INTEL_IFORT
   I_IFORT=$(INTEL_IFORT)
endif
```
and then try to compiler again. Do a `git clean` to remove the old files.

**Reply by lihy213** [reply]
Does it mean to change the lowercase `ifort` to `ifx`? But it appeared the same error as I mentioned before when I changed that. :(

**Reply by lihy213** [reply]
Like this,
```fortran

# set compiler type

I_IFORT=ifx
ifdef INTEL_IFORT
   I_IFORT=$(INTEL_IFORT)
endif

**Comment by mcgratta** [comment]
yes

**Reply by lihy213** [reply]
There's no way to solve it? 😞 
When I typed `make_fds.bat` within `impi_intel_win`, it met the error, conversely within `impi_intel_win_db`.

**Comment by marcosvanella** [comment]
Try this:

**Reply by marcosvanella** [reply]
INTEL_IFORT=ifx
I_IFORT=ifort
ifdef INTEL_IFORT
   I_IFORT=$(INTEL_IFORT)
endif

**Reply by lihy213** [reply]
Yes, I tried it, but something new happened. 😢 
![image](https://github.com/firemodels/fds/assets/96673442/2c056f8e-a590-4336-826c-d5715f663a68)

![image](https://github.com/firemodels/fds/assets/96673442/19fa2642-4e9b-494d-9cfa-aca743d81687)

**Reply by rmcdermo** [reply]
You have to be in a clone repository to build, not just a download of the tarball.

https://github.com/firemodels/fds/wiki/Git-Notes-Getting-Started

**Comment by lihy213** [comment]
> You have to be in a clone repository to build, not just a download of the tarball.
> 
> https://github.com/firemodels/fds/wiki/Git-Notes-Getting-Started

It doesn't work though, maybe I should reboot my PC again?

**Reply by rmcdermo** [reply]
What does not work?

**Reply by lihy213** [reply]
The same question after cloning the repo. BTW, I didn't modify the `makefile`.
![image](https://github.com/firemodels/fds/assets/96673442/e8d86af1-9180-4882-8ead-f890e5273751)

**Reply by rmcdermo** [reply]
It looks like you did not install the MKL libraries when you installed the Intel Compilers.

**Reply by lihy213** [reply]
Here, before I compiled FDS, the relevant libs were installed.
![image](https://github.com/firemodels/fds/assets/96673442/e539773d-9dbe-47a1-ae46-55f28bb5d630)
![image](https://github.com/firemodels/fds/assets/96673442/05b0a127-53f1-4e3c-bb89-8c08e8e95298)

**Comment by mcgratta** [comment]
The MKL libraries come with the HPC toolkit. You need the Base and HPC toolkits.

**Reply by lihy213** [reply]
Yes, I indeed installed these libs, cause the fortran file could be compiled in VS 2022.
![image](https://github.com/firemodels/fds/assets/96673442/a99e4603-2241-4020-b74d-34d7e19cdc83)
![image](https://github.com/firemodels/fds/assets/96673442/df6ece4a-2266-4f6f-8446-50ed8c5543fa)

**Comment by lihy213** [comment]
Anyone can help me? Great THANKS!

**Comment by rmcdermo** [comment]
We are trying.  But you are not giving us any more to go on.  You say you were able to compile with VS 2022?  Why change?  We are still using classic fortran to build FDS.

**Reply by lihy213** [reply]
I build FDS under the instruction of the guidance [Compiling FDS with Intel oneAPI](https://github.com/firemodels/fds/wiki/Compiling-FDS-with-Intel-oneAPI). I'm able to compile with VS 2022 means that intel Fortran and MKL libs(yes, you mean the Base and HPC toolkits) were installed correctly but had nothing to do with FDS. And it looks that nothing wrong with building FDS from the picture, changing ifort to ifx is just a reminder. The information indicates that something is wrong unexpectedly.
![image](https://github.com/firemodels/fds/assets/96673442/f920b23d-f917-41d6-8e0a-6adbd8dea90d)

**Reply by gforney** [reply]
Is your user name DELL? Try rebooting your pc. Can you delete the file the compiler failed to delete

**Reply by lihy213** [reply]
Yes! The user name is DELL. I reboot my PC again and then compiled `make_fds.bat`, but it still meets the same problem. I also delete the files manually, but it's different every time. 
![image](https://github.com/firemodels/fds/assets/96673442/84dfeaf1-ffef-4905-b423-11cbc5639628)

**Comment by rmcdermo** [comment]
Sorry, there is really very little we can do since we do not have access to your system.  The only think I can think of at this point would be to open the task manager while the compilation is running and see if you somehow overrun your memory.  I've never seen that happen at compile time, but I don't know what else to suggest.

**Reply by lihy213** [reply]
Thanks for your great help! It's a tough situation, maybe I should try another way. 💪

**Comment by mcgratta** [comment]
I just installed the oneAPI Fortran compiler 2024.0.2. When I compile with ifx I get an error.

**Reply by lihy213** [reply]
Yes, it's same to you.

**Comment by lihy213** [comment]
The problem has been solved! The reason was that the path of fds git files included Chinese characters. Too frustrated! 😭

**Comment by mcgratta** [comment]
Whenever I have trouble with a Windows application, the first thing I think about is whether the blank spaces in the folder names are a source of the problem. Add Chinese characters to your list.
